spring-cloud-stream-tuple/src/main/java/org/springframework/cloud/stream/tuple/TupleBuilder.java

```diff
@@ -68,8 +70,22 @@
 		defaultConversionService.addConverter(new StringToDateConverter(dateFormat));
 	}
 
+	/**
+	 * Return a new builder that will create immutable Tuples.
+	 */
+	public static TupleBuilder immutableTuple() {
+		return new TupleBuilder(false);
+	}
+
+	/**
+	 * Return a new builder that will create {@link MutableTuple}.
+	 */
 	public static TupleBuilder tuple() {
-		return new TupleBuilder();
+		return new TupleBuilder(true);
```

Comment:
The default should be immutable, as it was before, with mutable being the new option.